api/envoy/extensions/clusters/dns/v3/dns_cluster.proto

```diff
@@ -0,0 +1,89 @@
+syntax = "proto3";
+
+package envoy.extensions.clusters.dns.v3;
+
+import "envoy/config/core/v3/extension.proto";
+import "envoy/extensions/clusters/common/dns/v3/dns.proto";
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.clusters.dns.v3";
+option java_outer_classname = "DnsClusterProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/clusters/dns/v3;dnsv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: DNS cluster configuration]
+
+// Configuration for strict and logical DNS clusters.
+// [#extension: envoy.clusters.dns]
+
+// [#next-free-field: 10]
+message DnsCluster {
+  message RefreshRate {
+    // Specifies the base interval between refreshes. This parameter is required and must be greater
+    // than zero and less than
+    // :ref:`max_interval <envoy_v3_api_field_extensions.clusters.dns.v3.DnsCluster.RefreshRate.max_interval>`.
+    google.protobuf.Duration base_interval = 1 [(validate.rules).duration = {
+      required: true
+      gt {nanos: 1000000}
+    }];
+
+    // Specifies the maximum interval between refreshes. This parameter is optional, but must be
+    // greater than or equal to the
+    // :ref:`base_interval <envoy_v3_api_field_extensions.clusters.dns.v3.DnsCluster.RefreshRate.base_interval>`  if set. The default
+    // is 10 times the :ref:`base_interval <envoy_v3_api_field_extensions.clusters.dns.v3.DnsCluster.RefreshRate.base_interval>`.
+    google.protobuf.Duration max_interval = 2 [(validate.rules).duration = {gt {nanos: 1000000}}];
+  }
+
+  // This value is the cluster’s DNS refresh rate. The value configured must be at least 1ms.
+  // If this setting is not specified, the
+  // value defaults to 5000ms.
+  google.protobuf.Duration dns_refresh_rate = 3 [(validate.rules).duration = {gt {nanos: 1000000}}];
```

Comment:
Personally I think `dns` is now redundant in all of these field names and can be removed, but i don't feel strongly about it. Not sure if @markdroth has any opinions on this. Take it or leave it.

Reply:
I agree, removing the `dns_` prefix from these field names does make sense now.

Reply:
I kept it mostly out of convenience to interopt with

https://github.com/envoyproxy/envoy/blob/ab9ff8e5aa5735d3bb796bd27c44638c292c5593/source/common/config/utility.h#L438-L440

This function expects `T` to have a `[has_]dns_failure_refresh_rate()` method and is already templated with `envoy::config::cluster::v3::Cluster` and `envoy::extensions::common::dynamic_forward_proxy::v3::DnsCacheConfig`.

https://github.com/envoyproxy/envoy/blob/ab9ff8e5aa5735d3bb796bd27c44638c292c5593/source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc#L601-L603

so I chose to keep the `dns_` prefix in the proto to avoid rewriting the function.


Another reason is that it makes field names consistent across protos. For example, [`DnsCacheConfig`](https://github.com/envoyproxy/envoy/blob/main/api/envoy/extensions/common/dynamic_forward_proxy/v3/dns_cache.proto) also has the `dns_` prefix for most of its fields even though its redundant there too.

Continuing on with the consistency argument, if the goal is to deprecate `config.cluster.v3.Cluster.{dns_lookup_family,typed_dns_resolver_config}`, we're going to have to move those fields to the typed configurations that use them. That is, we would have to add `RedisClusterConfig.{dns_lookup_family,typed_dns_resolver_config} and it would be nice if all the fields had the same name for the sake of ergonomics and if we ever wanted to do templating shenanigans again.

Reply:
@mattklein123 @markdroth Thoughts on this?

Reply:
It's fine to keep it as is.

Reply:
Yeah, I'm okay with keeping this as-is too.